common/flatpak-dir.c

```diff
@@ -10155,6 +10155,34 @@ flatpak_dir_get_remote_collection_id (FlatpakDir *self,
   return collection_id;
 }
 
+static void
+remove_installed_refs (FlatpakDir *self,
+                       const char *remote,
+                       GPtrArray  *refs)
+{
+  int i;
+
+  for (i = refs->len - 1; i >= 0; i--)
+    {
+      const char *ref = g_ptr_array_index (refs, i);
+      g_autofree char *latest = NULL;
+
+      latest = flatpak_dir_read_latest (self, remote, ref, NULL, NULL, NULL);
+      if (latest == NULL)
+        continue;
+
+      g_autofree char *active = NULL;
+      active = flatpak_dir_read_active (self, ref, NULL);
+      if (active == NULL)
+        continue;
+
+      if (strcmp (active, latest) != 0)
```

Comment:
This implies that if the app has an undeployed update it's treated as uninstalled. That might make sense for the update command but less so for the install command. Perhaps we should use `flatpak_dir_get_if_deployed` instead? That's what is used in a similar situation in `common/flatpak-transaction.c`.

Reply:
I was going off what flatpak-builtins-install.c does for the no-pull case:

```
if (opt_no_pull)
    refs = flatpak_dir_find_local_refs (this_dir, this_remote, id, branch, this_default_branch, arch,
                                                        flatpak_get_default_arch (),
                                                        matched_kinds, FIND_MATCHING_REFS_FLAGS_FUZZY,
                                                        cancellable, &local_error);
else
  refs = flatpak_dir_find_remote_refs (this_dir, this_remote, id, branch, this_default_branch, arch,
                                                         flatpak_get_default_arch (),
                                                         matched_kinds, FIND_MATCHING_REFS_FLAGS_FUZZY,
                                                         cancellable, &local_error);
```

Reply:
I don't understand the connection. Why does the use of `flatpak_dir_find_local_refs` mean we shouldn't use `flatpak_dir_get_if_deployed`?

Reply:
Ny neither, tbh. The idea is to find local refs that are not installed.

Reply:
I confirmed that the way it's written now with `flatpak_dir_get_if_deployed` shows pulled-but-not-deployed refs as installable, as you'd expect.